packages/fxa-admin-panel/server/lib/csp/blocking.ts

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Middleware to take care of CSP. CSP headers are not sent unless config
+// option 'csp.enabled' is set (default true).
+
+import url from 'url';
+
+function getOrigin(link: string) {
+  const parsed = url.parse(link);
+  return `${parsed.protocol}//${parsed.host}`;
+}
+
+/**
+ * blockingCspMiddleware is where to declare rules that will cause a resource
+ * to be blocked if it runs afowl of a rule.
+ */
+export default function cspBlocking(config: { [key: string]: any }) {
+  const ADMIN_SERVER = getOrigin(config.get('servers.admin.url'));
+  const CDN_URL = config.get('staticResources.url');
+  const DATA = 'data:';
+  const PUBLIC_URL = config.get('listen.publicUrl');
+  const HOT_RELOAD_WEBSOCKET = PUBLIC_URL.replace(/^http/, 'ws');
+  const EXTRA_IMG_SRC = config.get('csp.extraImgSrc');
+
+  //
+  // Double quoted values
+  //
+  const NONE = "'none'";
+  const SELF = "'self'";
+  const UNSAFE_INLINE = "'unsafe-inline'";
```

Comment:
I know we chatted about this in Slack, but were you able to get this working without `unsafe-inline` styles?

Reply:
I was! What I also found out, though, was that I was not able to get it working without `unsafe-inline` on scripts. Any idea why that would be?

Reply:
Ahh I just assumed this was for `styles`, oops. We definitely don't want it on `scripts`. I'll have to check into your branch and poke at it because I'm not sure why it'd complain.